source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -31,6 +30,20 @@ DnsCacheImpl::DnsCacheImpl(
               config, refresh_interval_.count(), random)),
       host_ttl_(PROTOBUF_GET_MS_OR_DEFAULT(config, host_ttl, 300000)),
       max_hosts_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, max_hosts, 1024)) {
+  envoy::config::core::v3::DnsResolverOptions dns_resolver_options;
+  if (config.has_dns_resolver_options()) {
+    const auto& config_dns_options = config.dns_resolver_options();
```

Comment:
Same as above.
Also not sure if the two pieces of code can be refactored, as they are essentially the same.

Reply:
Took care of this change. Need your thoughts here. In the updated changes the code looks like
```
    envoy::config::core::v3::DnsResolverOptions dns_resolver_options;
    if (cluster.has_dns_resolver_options()) {
      dns_resolver_options.CopyFrom(cluster.dns_resolver_options());
    } else {
      // <code comment>
      dns_resolver_options.set_use_tcp_for_dns_lookups(cluster.use_tcp_for_dns_lookups());
    }
```
technically we don't need both the `if` & `else` condition. We could just do with 
``` 
  envoy::config::core::v3::DnsResolverOptions dns_resolver_options;
  dns_resolver_options.CopyFrom(cluster.dns_resolver_options());
  if (!cluster.has_dns_resolver_options()) {
    dns_resolver_options.set_use_tcp_for_dns_lookups(cluster.use_tcp_for_dns_lookups());
  }
```

But I thought it is better if we make it explicit that if legacy control plane has not configured `dns_resolver_options` then we stick to just setting `dns_resolver_options.use_tcp...` based on deprecated field `use_tcp_for_dns_lookups`.